libpcsxcore/psxbios.c

```diff
@@ -3078,6 +3080,21 @@ void psxBiosInit() {
 	psxHu32ref(0x1060) = SWAPu32(0x00000b88);
 
 	hleSoftCall = FALSE;
+	
+	/*	Some games like R-Types, CTR, Fade to Black read from adress 0x00000000 due to uninitialized pointers.
+		See Garbage Area at Address 00000000h in Nocash PSX Specfications for more information.
+		R-type will work if PsxM[0] is non-zero. (Meaning that it should not be 00000003h as he implies).
+		Crash Team Racing will refuse to boot if psxM[2] and psx[3] are not set to 0.
+		Fade to Black can crash upon memory card access if byte 5 is set to the wrong value.
+	*/
+	psxM[0] = SWAPu32(0x3C);
+	psxM[1] = SWAPu32(0x1A);
+	psxM[2] = SWAPu32(0x00);
+	psxM[3] = SWAPu32(0x00);
+	psxM[4] = SWAPu32(0x27);
+	psxM[5] = SWAPu32(0x5A);
+	psxM[6] = SWAPu32(0x0C);
+	psxM[7] = SWAPu32(0x80);
```

Comment:
You are writing bytes here, not u32, SWAPu32 is wrong and unneeded.
(not that we support big endian, but still)...

Reply:
Thanks, i think i've fixed this properly now.